### PR TITLE
Storage for preambles

### PR DESCRIPTION
**Pull request by cmc333333**
This adds very, very naive storage for notice preambles. Really, it's just carving out a namespace and allowing _any_ json to be written/read. We can tighten down the schema when we know more.

Builds on #36. See [diff](https://github.com/cmc333333/regulations-core/compare/more-refactors...cmc333333:12-preambles-rebase)

For eregs/notice-and-comment#12

### CODE REVIEW

File: regcore/migrations/0002_preamble.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
+# -*- coding: utf-8 -*-
```
Comment (jmcarp) [inline regcore/migrations/0002_preamble.py:1]: Should this migration be re-numbered to account for the migrations from #38? Not sure if this matters.

Comment (cmc333333) [inline regcore/migrations/0002_preamble.py:1]: No; the "merge" migration takes care of that.

File: regcore/tests/db_es_tests.py [hunk 2]
```diff
@@ -184,3 +185,18 @@ def test_put(self):
...
+            self.assertIsNone(ESPreambles().get('docdoc'))
+
+    def test_get_success(self):
+        with self.expect_get('preamble', 'docdoc', {'arbitrary': True}):
+            self.assertEqual(ESPreambles().get('docdoc'), {'arbitrary': True})
+
+    def test_put(self):
+        with self.expect_put('preamble', 'docdoc'):
+            ESPreambles().put('docdoc', {"some": "structure"})
+        self.assertEqual(self.call_args[0][2], {"some": "structure"})
```
Comment (jmcarp) [inline regcore/tests/db_es_tests.py]: I didn't notice this earlier, but I don't love storing `call_args` on `self`. I wonder if we could stick this on the context manager instead so that it can't leak across tests. Anyway, very minor.

Comment (cmc333333) [inline regcore/tests/db_es_tests.py]: Good point, can do.

File: regcore_write/views/preamble.py [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+from regcore.responses import success, user_error
+from regcore_write.views.security import secure_write
+
+
+@secure_write
+def add(request, docnum):
+    """Add the preamble to the db"""
+    try:
+        preamble = json.loads(request.body.decode('utf-8'))
+    except (ValueError, UnicodeError):
```
Comment (jmcarp) [inline regcore_write/views/preamble.py]: Another drive-by question. I don't know how Django handles this kind of thing, but it would be neat if we could move this repeated logic into a helper function that throws an exception if we can't parse, and that exception is caught by Django and turned into a 400 or 422 or whatever. It looks like we could raise `SuspiciousOperation`, since that gets translated into a 400, but that might not be semantically correct.

Basically, I'm looking for the Django equivalent of http://flask.pocoo.org/docs/0.10/patterns/apierrors/#registering-an-error-handler.

File: regcore/tests/db_es_tests.py (hunk 2 above)
Comment (jmcarp) [inline regcore/tests/db_es_tests.py]: https://s-media-cache-ak0.pinimg.com/236x/bf/75/e9/bf75e9a8bfd058039b9654bdecd3c714.jpg

**Comment by jmcarp** [comment]
I added a few marginally relevant questions on the diffs, but this looks good to merge.

**Comment by cmc333333** [comment]
@jmcarp updated! Thanks for the suggestions
